src/Parser/AfflictionWarlock/Modules/Features/UABuffTracker.js

```diff
@@ -0,0 +1,125 @@
+import React from 'react';
+
+import Module from 'Parser/Core/Module';
+import Enemies from 'Parser/Core/Modules/Enemies';
+
+import SPELLS from 'common/SPELLS';
+import SpellIcon from 'common/SpellIcon';
+import { formatPercentage } from 'common/format';
+import StatisticBox, { STATISTIC_ORDER } from 'Main/StatisticBox';
+
+import {UNSTABLE_AFFLICTION_DEBUFF_IDS} from '../../Constants';
+
+class UABuffTracker extends Module {
+  static dependencies = {
+    enemies: Enemies,
+  };
+
+  totalTicks = 0;
+  unbuffedTicks = 0;
+  ticksBuffedByReap = 0;
+  ticksBuffedByDrain = 0;
+  ticksBuffedByHaunt = 0;
+  ticksBuffedByBoth = 0;
+  hasMG = false;
+  hasHaunt = false;
+
+  on_initialized() {
+    if(!this.owner.error) {
+      this.hasMG = this.owner.selectedCombatant.hasTalent(SPELLS.MALEFIC_GRASP_TALENT.id);
+      this.hasHaunt = this.owner.selectedCombatant.hasTalent(SPELLS.HAUNT_TALENT.id);
+    }
+  }
+
+  on_byPlayer_damage(event) {
+    const spellId = event.ability.guid;
+    if(UNSTABLE_AFFLICTION_DEBUFF_IDS.indexOf(spellId) === -1)
```

Comment:
I'll have to think about it, but I don't really like skipping out on curly brackets on one line blocks because I believe it's easy to mess up and create a bug, harder to read, and violates jslint. It's no big deal though, you'll see the eslint rule appear or not when I make up my mind.